### PR TITLE
Application cci

### PR DESCRIPTION
**Pull request by RumiAust**
Please review my code.

### CODE REVIEW

**Comment by ifti24** [comment]
Although you have not attached or included your DB changes here, I have some suggestions for your DB script.

1. Place proper comment symbol in every line of your DB script. So that one can just copy and execute it without doing any changes. For example - 
`=======================
MST_PARAMETER_SETTING
=======================`
This is not a comment. One needs to either remove it or modify it. If you write it in the following way, it can help a developer.
`--=======================
--MST_PARAMETER_SETTING
--=======================`

**Comment by ifti24** [comment]
*My observations on your table structures*

1. APPLICATION_TES_QUESTIONS 

a. APPLICATION_TES_QUESTIONS can be TES_Questions
b. QUESTION_ID column seems redundant to me.
c. QUESTION_DETAILS can be only QUESTION
d. SEMESTER_ID : What does this field indicate?
e. APPLIED_ON: What does this field mean? And Why the datatype is TimeStamp? Why not Date? We should be consistently throughout the application.
f. I don't know whether any developer used TimeStamp data type in our application or not. But we should use one not multiple data type for representing the date time.

g. LAST_MODIFIED: Have u seen in most of our data table we placed a column named LAST_MODIFIED. Do you know its usages?

File: ums-api/src/main/java/org/ums/domain/model/mutable/MutableApplicationTES.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+  void setDesignation(final String pDesignation);
+
+  void setStatus(final Integer pStatus);
+
+  void setAppliedDate(final String pAppliedDate);
+
+  void setProgramShortName(final String pProgramShortName);
+
+  void setSemesterStartDate(final String pSemesterStartDate);
```
Comment (ifti24) [inline ums-api/src/main/java/org/ums/domain/model/mutable/MutableApplicationTES.java]: pSemesterStartDate. why not pSemesterEndDate?

File: ums-core/src/main/java/org/ums/persistent/dao/PersistentApplicationTESDao.java [hunk 2]
```diff
@@ -0,0 +1,749 @@
...
+      "select QUESTION_ID,SEMESTER_ID,STATUS from APPLICATION_TES_SET_QUESTIONS WHERE SEMESTER_ID=?";
+  String getSectionList =
+      "SELECT COURSE_ID,ASSIGNED_SECTION from TES_COURSE_ASSIGN WHERE COURSE_ID=? AND SEMESTER_ID=? AND TEACHER_ID=? ORDER BY ASSIGNED_SECTION";
+  String getAllSectionForSelectedCourse =
+      "select COURSE_ID,\"SECTION\" from COURSE_TEACHER WHERE COURSE_ID=? AND SEMESTER_ID=? AND TEACHER_ID=? ORDER BY \"SECTION\"";
+  String getDeptListByFacultyId = "SELECT DISTINCT  DEPT_ID FROM MST_PROGRAM WHERE CATEGORY_ID=?";
+  String getCourseForQuestionWiseReport =
+      "SELECT DISTINCT a.COURSE_ID FROM MST_COURSE a,COURSE_TEACHER b WHERE a.COURSE_TYPE=1 AND a.OFFER_BY=? AND a.\"YEAR\"=? AND a.SEMESTER=? AND b.SEMESTER_ID=? AND a.COURSE_ID=b.COURSE_ID";
+
+  String getTeacherListForQuestionWiseReport =
```
Comment (ifti24) [inline ums-core/src/main/java/org/ums/persistent/dao/PersistentApplicationTESDao.java]: Should it here?

File: ums-core/src/main/java/org/ums/persistent/dao/PersistentApplicationTESDao.java [hunk 3]
```diff
@@ -0,0 +1,749 @@
...
+    mJdbcTemplate.batchUpdate(INSERT_ONE,  parameters);
+    return  parameters.stream()
+            .map(paramArray  ->  (Long)  paramArray[0])
+            .collect(Collectors.toCollection(ArrayList::new));
+  }
+
+  @Override
+  public int delete(List<MutableApplicationTES> pMutableList) {
+    String query = DELETE_ALL + " WHERE QUESTION_ID=? AND SEMESTER_ID=?";
+    List<Object[]> parameters = deleteParamList(pMutableList);
```
Comment (ifti24) [inline ums-core/src/main/java/org/ums/persistent/dao/PersistentApplicationTESDao.java]: Why do we need semestr_id here? Isn't ID enough here?

File: ums-api/src/main/java/org/ums/enums/ApplicationStatus.java [hunk 4]
```diff
@@ -1,25 +1,49 @@
...
   WAITING_HEADS_APPROVAL(2, "Waiting for head's approval"),
   HEADS_ACCEPTED(3, "Allowed by the head"),
   HEADS_REJECTED(4, "Rejected by the head"),
   AAO_ACCEPTED(5, "Allowed by the AAO"),
-  AAO_REJECTED(6, "Rejected by the AAO");
+  AAO_REJECTED(6, "Rejected by the AAO"),
+  WAITING_FOR_PAYMENT(7, "Waiting for payment"),
+  APPROVED(8, "Approved"),
+  REJECTED(9, "Rejected"),
+  DEFAULT_LOAD_STATUS(10, "False"),
```
[the remark this answers is not in the document]
Comment (RumiAust) [reply ums-api/src/main/java/org/ums/enums/ApplicationStatus.java]: its for cci

File: ums-api/src/main/java/org/ums/enums/ApplicationStatus.java [hunk 5]
```diff
@@ -1,25 +1,49 @@
...
+
+import java.util.EnumSet;
+import java.util.HashMap;
+import java.util.Map;
+
 /**
  * Created by Morshed on 3/22/2016.
  */
+
 public enum ApplicationStatus {
```
[the remark this answers is not in the document]
Comment (RumiAust) [reply ums-api/src/main/java/org/ums/enums/ApplicationStatus.java]: Morsehd vaiya theke sunte hobe eta ektu

File: ums-api/src/main/java/org/ums/enums/FacultyType.java [hunk 6]
```diff
@@ -4,8 +4,9 @@
  * Created by Monjur-E-Morshed on 03-Dec-16.
  */
 public enum FacultyType {
```
[the remark this answers is not in the document]
Comment (RumiAust) [reply ums-api/src/main/java/org/ums/enums/FacultyType.java]: i didn't know vaiya eta already created

File: ums-core/src/main/java/org/ums/decorator/ApplicationTESDaoDecorator.java [hunk 7]
```diff
@@ -0,0 +1,197 @@
...
+    return getManager().getTeacherListForQuestionWiseReport(pCourseId, pSemesterId);
+  }
+
+  @Override
+  public List<ApplicationTES> getDeptListByFacultyId(Integer pFacultyId) {
+    return getManager().getDeptListByFacultyId(pFacultyId);
+  }
+
+  @Override
+  public List<ApplicationTES> getAllSectionForSelectedCourse(String pCourseId, String pTeacherId, Integer pSemesterId) {
```
[the remark this answers is not in the document]
Comment (RumiAust) [reply ums-core/src/main/java/org/ums/decorator/ApplicationTESDaoDecorator.java]: i didn't understand it vaiya

File: ums-core/src/main/java/org/ums/decorator/ApplicationTESDaoDecorator.java [hunk 8]
```diff
@@ -0,0 +1,197 @@
...
+    return getManager().addQuestions(pMutableList);
+  }
+
+  @Override
+  public Integer getTotalRegisteredStudentForCourse(String pCourseId, String pSection, Integer pSemesterId) {
+    return getManager().getTotalRegisteredStudentForCourse(pCourseId, pSection, pSemesterId);
+  }
+
+  @Override
+  public List<ApplicationTES> getDeptList() {
```
[the remark this answers is not in the document]
Comment (RumiAust) [reply ums-core/src/main/java/org/ums/decorator/ApplicationTESDaoDecorator.java]: i didn't know it vaiya

File: ums-core/src/main/java/org/ums/persistent/dao/PersistentApplicationTESDao.java (hunk 2 above)
Comment (RumiAust) [reply ums-core/src/main/java/org/ums/persistent/dao/PersistentApplicationTESDao.java]: Question wise Report e Parameters
1.Dept
2.year
3.Semester 
4.Semester Id
5.Question Id
eta tahole kothai dibo ami?

File: ums-core/src/main/java/org/ums/persistent/dao/PersistentApplicationTESDao.java (hunk 3 above)
Comment (RumiAust) [reply ums-core/src/main/java/org/ums/persistent/dao/PersistentApplicationTESDao.java]: eta tes_Set_Question Table er
Ekhen eki Questionid  fall17,Spring 17 e set hoyte pare
so only question id diye delet korle sob delet hoye jabe

**Pull request merged**